feat: allow subscribe to opcua events  BREAKING CHANGE: new input Plugin to subscribe to opcua node_ids and receive their events

frmoschner opened the pull request:
## Summary
This Plugin is useful to subscribe to OPC UA node ids to receive upcoming events on the subscribed nodes. 

## Checklist
- [x] No AI generated code was used in this PR

## Related issues
OPCUA Event Subscription #16275

resolves #

frmoschner commented:
I run readme_linter and it passes. Why does the Pipeline fail?
What are all the ci/circle tests? Why are they failing on other plugins?

Need help! :)

srebhan commented:
@frmoschner first of all thank you for your contribution! Let me clarify a few thinks not fully clear to me... 

Is there any reason to **not** fold this into the [OPCUA listener plugin](https://github.com/influxdata/telegraf/tree/master/plugins/inputs/opcua_listener)? You could easily add `event` sections there similar to the `nodes` ones or add event-type filter to the `nodes` section. Is there anything preventing this?
Having another plugin duplicating large parts of the code is not maintainable in the long-run...

frmoschner commented:
Hi @srebhan,

Thank you very much for your reply! 
I have adjusted a few things and created a new pull request #16300.
So far OPCUA Polling & Subscribing is separated from each other, and it would make sense at first sight to merge DataChange Subscriptions (opcua_listener) & Event Subscriptions.
However, in my opinion, the new plugin should not be integrated into the Listener plugin, as a plugin always has a task, and the listener_plugin takes care of the processing of DataChangeNotifcations while the new plugin handles EventNotifications. The structure of the subscription is very different and the listener_plugin uses its own subscribeClientConfig to create client and subscription, which cannot be used for my plugin. So if you were to merge the plugins, you would have to differentiate in many places in the code whether the subscription is a DataChange or an event subscription. I think it is much more user-friendly and easier to maintain if the plugins are separated. 
I have seen that there is already an opcua client (plugins/common/opcua/client.go), which I am using from now on. I have removed my own client manager. This means there are no more redundancies in the plugins.
